src/Init/Data/Nat/Basic.lean

```diff
@@ -974,16 +1075,24 @@ protected theorem sub_eq_iff_eq_add {c : Nat} (h : b ≤ a) : a - b = c ↔ a =
 protected theorem sub_eq_iff_eq_add' {c : Nat} (h : b ≤ a) : a - b = c ↔ a = b + c := by
   rw [Nat.add_comm, Nat.sub_eq_iff_eq_add h]
 
+/-! ## Mul sub distrib -/
+
 theorem mul_pred_left (n m : Nat) : pred n * m = n * m - m := by
   cases n with
   | zero   => simp
   | succ n => rw [Nat.pred_succ, succ_mul, Nat.add_sub_cancel]
 
-/-! ## Mul sub distrib -/
+theorem mul_sub_one_left (n m : Nat) : (n - 1) * m = n * m - m := by
+  cases n with
+  | zero   => simp
+  | succ n =>
+    rw [Nat.add_sub_cancel, add_one_mul, Nat.add_sub_cancel]
 
 theorem mul_pred_right (n m : Nat) : n * pred m = n * m - n := by
   rw [Nat.mul_comm, mul_pred_left, Nat.mul_comm]
 
+theorem mul_sub_one_right (n m : Nat) : n * (m - 1) = n * m - n := by
+  rw [Nat.mul_comm, mul_sub_one_left, Nat.mul_comm]
```

Comment:
These theorems are all named incorrectly to begin with. Can you change to `sub_one_mul` and `mul_sub_one`. You could rename the pred theorems too, but please use `@[deprecated (since := "2024-06-01")] abbrev mul_pred_left := @pred_mul`, etc.